controller/state.go

```diff
@@ -122,7 +122,7 @@ func (m *appStateManager) getRepoObjs(app *v1alpha1.Application, source v1alpha1
 	}
 	manifestInfo, err := repoClient.GenerateManifest(context.Background(), &apiclient.ManifestRequest{
 		Repo:              repo,
-		HelmRepos:         helmRepos,
```

Comment:
I would remote `Repo` field, since now we are passing all repositories and repo server just need to find matching one by URL

Reply:
sorry - don't think it works like that

Reply:
`repos` field now contains all repositories ( git and helm ). Repository server already filters give repos by type for helm applications. In the same way repo server might filter repo by URL. Is not it?

Reply:
https://github.com/argoproj/argo-cd/issues/2259